docs/julia.rst

```diff
@@ -0,0 +1,540 @@
+as many values as the expression evaluates to.
+
+
+The exact nature of the global state is unspecified for this high level
+description. The local state ``L`` is a mapping of identifiers ``i`` to values ``v``,
+denoted as ``L[i] = v``.
+
+For an identifier ``v``, let ``$v`` be the name of the identifier.
+
+We will use a destructuring notation for the AST nodes.
+
+.. code::
+
+    E(G, L, <{St1, ..., Stn}>: Block) =
+        let G1, L1, mode = E(G, L, St1, ..., Stn)
+        let L2 be a restriction of L1 to the identifiers of L
+        G1, L2, mode
+    E(G, L, St1, ..., Stn: Statement) =
+        if n is zero:
+            G, L
+        else:
+            let G1, L1, mode = E(G, L, St1)
+            if mode is regular then
+                E(G1, L1, St2, ..., Stn)
+            otherwise
+                G1, L1, mode
+    E(G, L, FunctionDefinition) =
+        G, L, regular
+    E(G, L, <let var1, ..., varn := rhs>: VariableDeclaration) =
+        E(G, L, <var1, ..., varn := rhs>: Assignment)
+    E(G, L, <let var1, ..., varn>: VariableDeclaration) =
+        let L1 be a copy of L where L1[$vari] = 0 for i = 1, ..., n
+        G, L1, regular
+    E(G, L, <var1, ..., varn := rhs>: Assignment) =
+        let G1, L1, v1, ..., vn = E(G, L, rhs)
+        let L2 be a copy of L1 where L2[$vari] = vi for i = 1, ..., n
+        G, L2, regular
+    E(G, L, <for { i1, ..., in } condition post body>: ForLoop) =
+        if n >= 1:
+            let G1, L1, mode = E(G, L, i1, ..., in)
+            // mode has to be regular due to the syntactic restrictions
+            let G2, L2, mode = E(G1, L1, for {} condition post body)
+            // mode has to be regular due to the syntactic restrictions
+            let L3 be the restriction of L2 to only variables of L
+            G2, L3
+        else:
+            let G1, L1, v = E(G, L, condition)
+            if v is false:
+                G1, L1, regular
+            else:
+                let G2, L2, mode = E(G1, L, body)
+                if mode is break:
+                    G2, L2, regular
+                else:
+                    G3, L3, mode = E(G2, L2, post)
+                    E(G3, L3, for {} condition post body)
+    E(G, L, break: BreakContinue) =
+        G, L, break
+    E(G, L, continue: BreakContinue) =
+        G, L, continue
+
+    E(G, L, <name>: Identifier) =
+        G, L, regular, L[$name]
+    E(G, L, <fname(arg1, ..., argn)>: FunctionCall) =
+        G1, L1, vn = E(G, L, argn)
+        ...
+        G(n-1), L(n-1), v2 = E(G(n-2), L(n-2), arg2)
+        Gn, Ln, v1 = E(G(n-1), L(n-1), arg1)
+        Let <function fname (param1, ..., paramn) -> ret1, ..., retm block>
+        be the function of name $fname visible at the point of the call.
+        Let L' be a new local state such that
+        L'[$parami] = vi and L'[$reti] = 0 for all i.
+        Let G'', L'', rv1, ..., rvm = E(Gn, L', block)
+        G'', Ln, rv1, ..., rvm
```

Comment:
Well if I understand this correctly, E(Gn, L', block) is defined to return something like G'', L'', regular and rv1 ... rvm should be defined rvi = L''[$reti]